autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,334 @@
+# Slowdown bug not experienced on Linux if 'torch.multiprocessing.set_sharing_strategy('file_system')' commented out
+# NOTE: If below line is commented out, Torch uses many file descriptors. If issues arise, increase ulimit through 'ulimit -n 2048' or larger. Default on Linux is 1024.
+# torch.multiprocessing.set_sharing_strategy('file_system')
+
+# MacOS issue: torchvision==0.7.0 + torch==1.6.0 can cause segfaults; use torch==1.2.0 torchvision==0.4.0
+
+LABEL = '__label__'
+
+logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def make_temp_directory():
+    temp_dir = tempfile.mkdtemp()
+    try:
+        yield temp_dir
+    finally:
+        shutil.rmtree(temp_dir)
+
+
+# TODO: Takes extremely long time prior to training start if many (10000) continuous features from ngrams, debug - explore TruncateSVD option to reduce input dimensionality
+class NNFastAiTabularModel(AbstractModel):
+    """ Class for fastai v1 neural network models that operate on tabular data.
+
+        Attributes:
+            y_scaler: on a regression problems, the model can give unreasonable predictions on unseen data.
+            This attribute allows to pass a scaler for y values to address this problem. Please note that intermediate
+            iteration metrics will be affected by this transform and as a result intermediate iteration scores will be
+            different from the final ones (these will be correct).
+            https://scikit-learn.org/stable/modules/classes.html#module-sklearn.preprocessing
+
+        Hyperparameters:
+            'layers': list of hidden layers sizes; None - use model's heuristics; default is None
+
+            'emb_drop': embedding layers dropout; defaut is 0.1
+
+            'ps': linear layers dropout - list of values applied to every layer in `layers`; default is [0.1]
+
+            'bs': batch size; default is 256
+
+            'lr': maximum learning rate for one cycle policy; default is 1e-2;
+            see also https://fastai1.fast.ai/train.html#fit_one_cycle, One-cycle policy paper: https://arxiv.org/abs/1803.09820
+
+            'epochs': number of epochs; default is 30
+
+            # Early stopping settings. See more details here: https://fastai1.fast.ai/callbacks.tracker.html#EarlyStoppingCallback
+            'early.stopping.min_delta': 0.0001,
+            'early.stopping.patience': 10,
+
+            'smoothing': If > 0, then use LabelSmoothingCrossEntropy loss function for binary/multi-class classification;
+            otherwise use default loss function for this type of problem; default is 0.0.
+            See: https://docs.fast.ai/layers.html#LabelSmoothingCrossEntropy
+    """
+
+    model_internals_file_name = 'model-internals.pkl'
+    unique_category_str = '!missing!'
+
+    def __init__(self, path: str, name: str, problem_type: str, eval_metric=None, num_classes=None, stopping_metric=None, model=None, hyperparameters=None,
```

Comment:
can just be `super().__init__(**kwargs)`, and have `def __init__(self, **kwargs):`

Then `self.max_unique_categorical_values` can be `self.params.get('max_unique_categorical_values', 10000)`

Although ideally self.max_unique_categorical_values shouldn't exist, and self.params should be populated by this default value if necessary.

Reply:
updated